File: config/defaults/default.rb
#!/usr/bin/env ruby
# encoding: ascii-8bit
require 'pry'
require 'pwn'        # https://github.com/peter50216/pwntools-ruby
require 'heapinfo'   # https://github.com/david942j/heapinfo
require 'one_gadget' # https://github.com/david942j/one_gadget

host, port = '', 31337
@local = false
@p = ''
if ARGV.empty?
  host = '127.0.0.1'; @local = true
  # $z = Tubes::Process.new(@p)
else
  raise ArgumentError, 'host not set' if host.empty?
end
$z = Sock.new(host, port)
def z;$z;end
def h;@h ||= heapinfo(@p);end
def elf; @elf ||= ELF.new(@p); end
def libc; @libc ||= ELF.new('./libc.so.6', checksec: false); end
def debug!; context.log_level = :debug; end
#================= Exploit Start ====================
context.arch = :amd64
# debug!

z.interact

File: config/defaults/default.py
#!/usr/bin/env python2

from pwn import *
import sys
import os

host, port = '', 31337
local = False
if len(sys.argv) == 1:
  host = '127.0.0.1'
  local = True
r = remote(host, port)

context.arch = 'amd64'
#  context.log_level = 'debug'

r.interactive()
